plugins/uast-kotlin-base/src/org/jetbrains/uast/kotlin/KotlinAbstractUElement.kt

```diff
@@ -5,43 +5,39 @@
 
 package org.jetbrains.uast.kotlin
 
-import com.intellij.psi.PsiElement
-import org.jetbrains.uast.UElement
-import org.jetbrains.uast.UExpression
+import com.intellij.openapi.components.ServiceManager
+import org.jetbrains.uast.*
 import org.jetbrains.uast.kotlin.internal.KotlinUElementWithComments
 
-abstract class FirKotlinAbstractUElement(
-    private val givenParent: UElement?
+abstract class KotlinAbstractUElement(
+    givenParent: UElement?
 ) : KotlinUElementWithComments {
 
+    protected val languagePlugin: UastLanguagePlugin? by lz {
+        psi?.let { UastFacade.findPlugin(it) }
+    }
+
+    protected val resolveProviderService: KotlinUastResolveProviderService? by lz {
```

Comment:
Any sense to keep it nullable?

Reply:
I wondered this `project`-based lookup could fail, if, e.g., `psi` is `null` (for some Kotlin-specific declarations), its `project` is not setting (which sounds quite bad), or simply service is not configured well. It looks like, except for declaration expression, that's not the case, so I can make it non-nullable eventually.